main.rb

```diff
@@ -0,0 +1,132 @@
+# frozen_string_literal: false
+
+module Enumerable
+  def my_each(array)
+    num = []
+    x = 0
+    while x < array.length
+      num << yield(array[x])
+      x += 1
+    end
+    num
+  end
+
+  def my_each_with_index(array)
+    num = []
+    x = 0
+    while x < array.length
+      num << yield(array[x], x)
+      x += 1
+    end
+    num
+  end
+
+  def my_select(array)
+    num = []
+    x = 0
+    while x < array.length
+      if yield(array[x])
+        num << array[x]
+      end
+      x += 1
+    end
+    num
+  end
+
+  def my_all?(array)
+    num = true
+    x = 0
+    while x < array.length
+      unless yield(array[x])
+        num = false
+      end
+      x += 1
+    end
+    num
+  end
```

Comment:
-  Passes each element of the collection to the given block. The method returns true if the block never returns false or nil. If the block is not given, Ruby adds an implicit block of { |obj| obj } which will cause all? to return true when none of the collection members are false or nil.

If instead a pattern is supplied, the method returns whether pattern === element for every collection member.


The definition above is gotten from this [doc](https://ruby-doc.org/core-2.6.5/Enumerable.html)  This is how to go about it for each method in this module. It describes how to interact with the enumerator.

```  
def my_all?(pattern = nil)
      result = true
      if block_given?
          my_each { |ele| result &= (yield ele) }
      elsif pattern
          my_each { |ele| result &= pattern === ele }
      else
          my_each { |ele| result &= ele }
      end
      result
  end  
```

Go through the implementation above, understand it, as it would be a guide for the next methods my_none? and my_any?